config/configmodels/configmodels.go

```diff
@@ -43,12 +51,70 @@ type Config struct {
 // Type is the component type as it is used in the config.
 type Type string
 
+// EntityName is a component type and an optional name.
+type EntityName struct {
+	Type     Type
+	Name     string
+	fullName string
```

Comment:
You cannot expose the fields because changing one of the Type/Name would not change the fullName.

Reply:
Are you saying Type() and Name() should be getters? If so it's kind of an anti-pattern in Go.

Reply:
Then you have to always calculate the full name